engines/wintermute/base/gfx/x/modelx.cpp

```diff
@@ -371,120 +373,77 @@ bool ModelX::isTransparentAt(int x, int y) {
 		return false;
 	}
 
-	x += _lastOffsetX;
-	y += _lastOffsetY;
+	Math::Ray ray = _gameRef->_renderer3D->rayIntoScene(x, y);
 
-	//	if (!Rend->m_Camera) {
-	//		return true;
-	//	}
+	// transform to model space
+	Math::Vector3d end = ray.getOrigin() + ray.getDirection();
+	Math::Matrix4 m = _lastWorldMat;
+	m.inverse();
+	m.transpose();
+	m.transform(&ray.getOrigin(), true);
+	m.transform(&end, true);
+	Math::Vector3d pickRayDirection = end - ray.getOrigin();
 
-//	float resWidth, resHeight;
-//	float layerWidth, layerHeight;
-	float modWidth = 0.0f;
-	float modHeight = 0.0f;
-	bool customViewport = false;
-	//	Rend->GetProjectionParams(&ResWidth, &ResHeight, &LayerWidth, &LayerHeight, &ModWidth, &ModHeight, &CustomViewport);
+	return !_rootFrame->pickPoly(&ray.getOrigin(), &pickRayDirection);
+}
 
-	x -= _drawingViewport.left + modWidth;
-	y -= _drawingViewport.top + modHeight;
+//////////////////////////////////////////////////////////////////////////
+void ModelX::updateBoundingRect() {
+	_BBoxStart = Math::Vector3d(0, 0, 0);
+	_BBoxStart = Math::Vector3d(0, 0, 0);
 
-	if (customViewport) {
-		x += _gameRef->_renderer->_drawOffsetX;
-		y += _gameRef->_renderer->_drawOffsetY;
+	if (_rootFrame) {
+		_rootFrame->getBoundingBox(&_BBoxStart, &_BBoxEnd);
 	}
 
-	Math::Vector3d pickRayDir;
-	Math::Vector3d pickRayOrig;
+	_boundingRect.left = INT_MAX;

```

Comment:
same as above too limits.h

Reply:
REPLACED it by INT_MAX_VALUE and new constant INT_MIN_VALUE. Assuming two complement is ok? (so INT_MIN_VALUE = -INT_MAX_VALUE -1)